packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -476,35 +441,134 @@ export function createRouteManifest(
 			generate,
 			pathname: pathname || void 0,
 			prerender: false,
-			redirect: to,
-			redirectRoute: routes.find((r) => r.route === to),
+			redirect: 'status' in to
+				? {
+					status: to.status,
+					destination: to.destination,
+				}
+				: to.destination,
+			redirectRoute: routeMap.get(to.destination),
 			fallbackRoutes: [],
-		};
+		});
+	});
 
-		const lastSegmentIsDynamic = (r: RouteData) => !!r.segments.at(-1)?.at(-1)?.dynamic;
+	return routes;
+}
 
-		const redirBase = path.posix.dirname(route);
-		const dynamicRedir = lastSegmentIsDynamic(routeData);
-		let i = 0;
-		for (const existingRoute of routes) {
-			// An exact match, prefer the page/endpoint. This matches hosts.
-			if (existingRoute.route === route) {
-				routes.splice(i + 1, 0, routeData);
-				return;
+function routesCollide(a: RouteData, b: RouteData) {
```

Comment:
It's worth adding a comment to the function that explains what this function is in charge of, and a brief explanation of its business logic.